crates/sui-indexer-alt-jsonrpc/src/api/dynamic_fields/response.rs

```diff
@@ -2,86 +2,144 @@
-#[rpc(server, namespace = "suix")]
-trait DynamicFieldsApi {
-    /// Return the information from a dynamic field based on its parent ID and name.
-    #[method(name = "getDynamicFieldObject")]
-    async fn get_dynamic_field_object(
-        &self,
-        /// The ID of the parent object
-        parent_object_id: ObjectID,
-        /// The Name of the dynamic field
-        name: DynamicFieldName,
-    ) -> RpcResult<SuiObjectResponse>;
-}
+use super::error::Error;
 
-pub struct DynamicFields(pub Context);
+/// Fetch the latest version of a dynamic field object, identified by its parent ID and name.
+pub(super) async fn dynamic_field_object(
+    ctx: &Context,
+    parent_object_id: ObjectID,
+    name: DynamicFieldName,
+) -> Result<SuiObjectResponse, RpcError<Error>> {
+    let layout = resolve_type(ctx, &name.type_).await?;
+    let bytes = SuiJsonValue::new(name.value)
+        .map_err(|e| invalid_params(Error::BadName(e)))?
+        .to_bcs_bytes(&layout)
+        .map_err(|e| invalid_params(Error::TypeMismatch(name.type_.clone(), e)))?;
 
-#[derive(thiserror::Error, Debug)]
-enum Error {
-    #[error("Bad dynamic field name: {0}")]
-    BadName(anyhow::Error),
+    let df = load_df(ctx, parent_object_id, &name.type_, &bytes);
+    let dof = load_dof(ctx, parent_object_id, &name.type_, &bytes);
+    let (df, dof) = try_join!(df, dof)
+        .with_context(|| format!("Failed to fetch dynamic field on {parent_object_id}"))?;
 
-    #[error("Invalid type {0}: {1}")]
-    BadType(TypeTag, sui_package_resolver::error::Error),
+    let Some(object) = df.or(dof) else {
+        return Ok(SuiObjectResponse::new_with_error(
+            SuiObjectResponseError::DynamicFieldNotFound { parent_object_id },
+        ));
+    };
 
-    #[error("Could not serialize dynamic field name as {0}: {1}")]
-    TypeMismatch(TypeTag, anyhow::Error),
-}
+    let options = SuiObjectDataOptions::full_content();
 
-#[async_trait::async_trait]
-impl DynamicFieldsApiServer for DynamicFields {
-    async fn get_dynamic_field_object(
-        &self,
-        parent_object_id: ObjectID,
-        name: DynamicFieldName,
-    ) -> RpcResult<SuiObjectResponse> {
-        let Self(ctx) = self;
-        Ok(dynamic_field_object_response(ctx, parent_object_id, name).await?)
-    }
+    use RpcError as E;
+    Ok(SuiObjectResponse::new_with_data(
+        objects::response::object_data_with_options(ctx, object, &options)
+            .await
+            .map_err(|e| match e {
+                E::InvalidParams(e) => match e {},
+                E::InternalError(e) => E::InternalError(e),
+            })?,
+    ))
 }
 
-impl RpcModule for DynamicFields {
-    fn schema(&self) -> Module {
-        DynamicFieldsApiOpenRpc::module_doc()
-    }
+/// Fetch the latest version of the object identified by `object_id`, treat it as if it is a
+/// `sui::dynamic_field::Field<K, V>`, and extract the name and value from it.
+pub(super) async fn dynamic_field_info(
```

Comment:
This is the main new function -- the body is similar to how the same functionality is implemented for `sui-indexer`:

https://github.com/MystenLabs/sui/blob/6fb9bda12442b71c853e7cff7eb14174ffd66253/crates/sui-indexer/src/indexer_reader.rs#L1160